src/os/bluestore/BlueFS.cc

```diff
@@ -339,7 +341,9 @@ void BlueFS::_add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   block_all[id].insert(offset, length);
 
   if (id < alloc.size() && alloc[id]) {
-    log_t.op_alloc_add(id, offset, length);
+    if (!skip)
```

Comment:
@yanghonggang - I'm not sure I understand what happens after new extent isn't preserved in the log - how  will new device obtain this information after successful attachment?

Reply:
> @yanghonggang - I'm not sure I understand what happens after new extent isn't preserved in the log - how will new device obtain this information after successful attachment?

new device's extent info will be record to the new log in _rewrite_log_and_layout_sync() -> _compact_log_dump_metadata():

```cpp
  t->op_init();
  for (unsigned bdev = 0; bdev < MAX_BDEV; ++bdev) {
    interval_set<uint64_t>& p = block_all[bdev];
    for (interval_set<uint64_t>::iterator q = p.begin(); q != p.end(); ++q) {
      auto bdev_new = bdev;
      ...
      dout(20) << __func__ << " op_alloc_add " << bdev_new << " 0x"
               << std::hex << q.get_start() << "~" << q.get_len() << std::dec
               << dendl;
      t->op_alloc_add(bdev_new, q.get_start(), q.get_len());
    }
  }
```
And only [this case](https://tracker.ceph.com/issues/44774) need to set skip to true.

Reply:
ping @ifed01 

how to reproduce this problem: https://tracker.ceph.com/issues/44774?issue_count=65&issue_position=3&next_issue_id=44757&prev_issue_id=44818#How-to-reproduce-this-problem